test/new.jl

```diff
@@ -2727,7 +2727,7 @@ end
 #
 # Note: these tests should be run on clean depots
 for v in (nothing, "true")
-    withenv("JULIA_PKG_USE_CLI_GIT" => v, "GIT_TERMINAL_PROMPT" => 0) do
+    withenv("JULIA_PKG_USE_CLI_GIT" => v, "GIT_TERMINAL_PROMPT" => 0, "GIT_ASKPASS" => "true") do
```

Comment:
```suggestion
    # On CI when JULIA_PKG_USE_CLI_GIT=true we need to tell the cli git to not prompt for credentials
    # GIT_ASKPASS=true forces the credential provider to return "" https://stackoverflow.com/a/71057440
    # GIT_TERMINAL_PROMPT=0 is also supposed to avoid the prompt but doesn't reliably
    withenv("JULIA_PKG_USE_CLI_GIT" => v, "GIT_TERMINAL_PROMPT" => 0, "GIT_ASKPASS" => "true") do
```